File: fixed_heap_dynamic.cpp
#include <bits/stdc++.h>
using namespace std;

// Fixed Heap Dynamic: Allocate fixed size using 'new' (heap memory)  size of the array is known at compile time (fixed)
int main() {
    int *arr = new int[5]; 
    cout << "Enter 6 elements:" << endl;
    for(int i = 0; i < 6; i++) {
        cin >> arr[i];
    }
    cout << "Fixed Heap Dynamic Array Elements: ";
    for(int i = 0; i < 6; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
    return 0;
}

File: fixed_heap_dynamic.js
// Fixed Heap Dynamic: Allocating fixed array (simulating heap in JS)
let arr = new Array(5).fill(0); // Fixed size 5, filled with 0

// Example: Fill the array
for (let i = 0; i < 5; i++) {
    arr[i] = i + 1;
}

console.log("Fixed Heap Dynamic Array Elements:");
console.log(arr);
